docs/contributing.rst

```diff
@@ -424,93 +416,82 @@ published and the test runs are updated to use the new stable version.
 Releasing
 ~~~~~~~~~
 
-Final Test Before Each Release
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Releases are typically done from the ``main`` branch, except when releasing a beta (in
+which case the beta is released from ``main``, and the previous stable branch is
+released from said branch).
 
-Before releasing a new version, build and test the package that will be released.
-There's a script to build and install the wheel locally, then generate a temporary
-virtualenv for smoke testing:
+Final test before each release
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Before releasing a new version, build and test the package that will be released:
 
 .. code:: sh
 
     $ git checkout main && git pull
+    $ make package-test
 
-    $ make package
-
-    # in another shell, navigate to the virtualenv mentioned in output of ^
-
-    # load the virtualenv with the packaged web3.py release
-    $ source package-smoke-test/bin/activate
-
-    # smoke test the release
-    $ pip install ipython
-    $ ipython
-    >>> from web3 import Web3, IPCProvider
-    >>> w3 = Web3(IPCProvider(provider_url))
-    >>> w3.is_connected()
-    >>> ...
-
+This will build the package and install it in a temporary virtual environment. Follow
+the instructions to activate the venv and test whatever you think is important.
 
-Verify The Latest Documentation
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-To preview the documentation that will get published:
+Review the documentation that will get published:
 
 .. code:: sh
 
     $ make docs
 
-
-Preview The Release Notes
-^^^^^^^^^^^^^^^^^^^^^^^^^
+Validate and preview the release notes:
 
 .. code:: sh
 
-   $ towncrier build --draft
+    $ make validate-newsfragments
 
+Build the release notes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Compile The Release Notes
-^^^^^^^^^^^^^^^^^^^^^^^^^
-
-After confirming that the release package looks okay, compile the release notes:
+Before bumping the version number, build the release notes. You must include the part of
+the version to bump (see below), which changes how the version number will show in the
+release notes.
 
 .. code:: sh
 
     $ make notes bump=$$VERSION_PART_TO_BUMP$$
 
+If there are any errors, be sure to re-run make notes until it works.
 
-You may need to fix up any broken release note fragments before committing. Keep
-running ``make build-docs`` until it passes, then commit and carry on.
+Push the release to github & pypi
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
I think these will cause some warnings? Haven't verified but would be nice to trim this at the source if it's coming from the template.

Reply:
False alarm. No warnings locally. 🚢